packages/kilt-did/src/DidDetails/LightDidDetails.spec.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+/**
+ * @group unit/did
+ */
+
+import { encodeAddress } from '@kiltprotocol/utils/src/Crypto'
+import type { IServiceDetails } from '@kiltprotocol/types'
+import { hexToU8a } from '@polkadot/util'
+import {
+  getEncodingForSigningKeyType,
+  getKiltDidFromIdentifier,
+} from '../Did.utils'
+import { LightDidDetails, LightDidDetailsCreationOpts } from './LightDidDetails'
+import { serializeAndEncodeAdditionalLightDidDetails } from './LightDidDetails.utils'
+import type { INewPublicKey } from '../types'
+
+describe('Light DID v1 tests', () => {
+  const authPublicKey = hexToU8a(
+    '0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
+  )
+  const encPublicKey = hexToU8a(
+    '0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb'
+  )
+  const testDid = getKiltDidFromIdentifier(
+    getEncodingForSigningKeyType('ed25519').concat(
+      encodeAddress(authPublicKey, 38)
+    ),
+    'light',
+    LightDidDetails.LIGHT_DID_VERSION
+  )
+  const authenticationDidKeyDetails: INewPublicKey = {
+    publicKey: authPublicKey,
+    type: 'ed25519',
+  }
+  let encryptionDidKeyDetails: INewPublicKey | undefined
+  let services: IServiceDetails[] | undefined
+
+  it('creates LightDidDetails from authentication key, encryption key, and service endpoints', () => {
+    encryptionDidKeyDetails = {
+      publicKey: encPublicKey,
+      type: 'x25519',
+    }
+    services = [
+      {
+        id: `service1`,
+        type: 'messaging',
+        serviceEndpoint: 'example.com',
+      },
+      {
+        id: `service2`,
+        type: 'telephone',
+        serviceEndpoint: '123344',
+      },
+    ]
+
+    const didCreationDetails: LightDidDetailsCreationOpts = {
+      authenticationKey: authenticationDidKeyDetails,
+      encryptionKey: encryptionDidKeyDetails,
+      services,
+    }
+
+    const encodedAdditionalDetails = serializeAndEncodeAdditionalLightDidDetails(
+      didCreationDetails
+    )!
+
+    const did = new LightDidDetails(didCreationDetails)
+    expect(did.did).toEqual(testDid.concat(':', encodedAdditionalDetails))
```

Comment:
S: Since this shouldn't break in the future, I think we should test it against the final string instead of relying on the `serializeAndEncodeAdditionalLightDidDetails` function, especially since it is used in the constructor for `LightDidDetails`

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/408/commits/ed4d9c3448b4bfc329e87bdd5930b88f88287e64.